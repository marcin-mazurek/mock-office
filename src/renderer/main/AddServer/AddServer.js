import React from 'react';
import { connect } from 'react-redux';
import { requestAdd } from '../../servers/actions';

class AddServer extends React.Component {
  constructor(props) {
    super(props);
    this.submit = this.submit.bind(this);

    this.state = {
      name: 'Server name',
      port: 3000
    };
  }

  handleNameChange(e) {
    this.setState({
      name: e.currentTarget.value
    });
  }

  handleChangePort(e) {
    this.setState({
      port: e.currentTarget.value
    });
  }

  submit() {
    this.props.requestAdd(this.state.name, this.state.port);
  }

  render() {
    return (
      <div>
        <label htmlFor="name">Name:</label>
        <input
          name="name"
          type="text"
          value={this.state.name}
          onChange={this.handleNameChange}
        />
        <label htmlFor="port">Port:</label>
        <input
          type="text"
          value={this.state.port}
          name="port"
          onChange={this.handleChangePort}
        />
        <button onClick={this.submit}>Create</button>
      </div>
    );
  }
}

AddServer.propTypes = {
  requestAdd: React.PropTypes.func.isRequired
};

const mapDispatchToProps = {
  requestAdd
};

export default connect(null, mapDispatchToProps)(AddServer);
