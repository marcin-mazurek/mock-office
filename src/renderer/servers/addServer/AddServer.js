import React from 'react';
import AddServerFormConnect from './AddServerForm';

const AddServer = () => (
  <div>
    <div className="view-header add-server__header">
      Add server
    </div>
    <AddServerFormConnect />
  </div>
);

export default AddServer;
