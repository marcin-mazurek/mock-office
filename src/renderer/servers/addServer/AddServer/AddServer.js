import React from 'react';
import { connect } from 'react-redux';
import init from '../actions';

class AddServer extends React.Component {
  constructor(props) {
    super(props);
    this.submit = this.submit.bind(this);
    this.handleNameChange = this.handleNameChange.bind(this);
    this.handleChangePort = this.handleChangePort.bind(this);
    this.handleTypeChange = this.handleTypeChange.bind(this);

    this.state = {
      name: 'Server name',
      port: 3000,
      type: 'http'
    };
  }

  handleNameChange(e) {
    this.setState({
      name: e.currentTarget.value
    });
  }

  handleChangePort(e) {
    this.setState({
      port: parseInt(e.currentTarget.value, 10)
    });
  }

  handleTypeChange(e) {
    this.setState({
      type: e.currentTarget.value
    });
  }

  submit() {
    this.props.add(this.state.name, this.state.port, this.state.type);
  }

  render() {
    return (
      <div>
        <label htmlFor="name">Name:</label>
        <input
          name="name"
          type="text"
          value={this.state.name}
          onChange={this.handleNameChange}
        />
        <label htmlFor="port">Port:</label>
        <input
          type="text"
          value={this.state.port}
          name="port"
          onChange={this.handleChangePort}
        />
        <label htmlFor="type">Type:</label>
        <select name="type" value={this.state.type} onChange={this.handleTypeChange}>
          <option value="http">http</option>
          <option value="ws">ws</option>
        </select>
        <button onClick={this.submit}>Create</button>
      </div>
    );
  }
}

AddServer.propTypes = {
  add: React.PropTypes.func.isRequired
};

const mapDispatchToProps = {
  add: init
};

export default connect(null, mapDispatchToProps)(AddServer);
