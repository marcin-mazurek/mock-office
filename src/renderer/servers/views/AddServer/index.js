import React from 'react';
import AddServerForm from '../../addServer/AddServerForm';

const AddServer = () => (
  <div>
    <header className="mdl-layout__header">
      <div className="mdl-layout__header-row">
        <span className="mdl-layout-title">
          Add server:
        </span>
      </div>
    </header>
    <AddServerForm />
  </div>
);

export default AddServer;
