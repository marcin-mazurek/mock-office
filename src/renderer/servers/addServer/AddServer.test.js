import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import AddServer from './AddServer';

describe('AddServer', () => {
  test('default snapshot', () => {
    const wrapper = shallow(<AddServer />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
