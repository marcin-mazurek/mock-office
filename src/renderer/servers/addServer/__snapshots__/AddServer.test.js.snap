exports[`AddServer  should render 1`] = `
<div>
  <Connect(AddServerForm) />
</div>
`;

exports[`AddServer de 1`] = `
<div>
  <Connect(AddServerForm) />
</div>
`;

exports[`AddServer defa 1`] = `
<div>
  <Connect(AddServerForm) />
</div>
`;

exports[`AddServer defau 1`] = `
<div>
  <Connect(AddServerForm) />
</div>
`;

exports[`AddServer default  1`] = `
<div>
  <Connect(AddServerForm) />
</div>
`;

exports[`AddServer default s 1`] = `
<div>
  <Connect(AddServerForm) />
</div>
`;

exports[`AddServer default sn 1`] = `
<div>
  <Connect(AddServerForm) />
</div>
`;

exports[`AddServer default snap 1`] = `
<div>
  <Connect(AddServerForm) />
</div>
`;

exports[`AddServer default snapshot 1`] = `
<div>
  <Connect(AddServerForm) />
</div>
`;

exports[`test  should render 1`] = `
<div>
  <Connect(AddServerForm) />
</div>
`;

exports[`test AddServer should render 1`] = `
<div>
  <Connect(AddServerForm) />
</div>
`;
